rasa/nlu/classifiers/fallback_classifier.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, List, Type, Text, Dict, Union
+
+from rasa.constants import DEFAULT_NLU_FALLBACK_INTENT_NAME
+from rasa.core.constants import DEFAULT_NLU_FALLBACK_THRESHOLD
+from rasa.nlu.classifiers.classifier import IntentClassifier
+from rasa.nlu.components import Component
+from rasa.nlu.training_data import Message
+from rasa.nlu.training_data.message import (
+    INTENT_RANKING_KEY,
+    INTENT_KEY,
+    INTENT_CONFIDENCE_KEY,
+)
+
+
+THRESHOLD_KEY = "threshold"
+FALLBACK_INTENT_NAME_KEY = "fallback_intent_name"
+
+
+class FallbackClassifier(Component):
+
+    # please make sure to update the docs when changing a default parameter
+    defaults = {
+        # ## Architecture of the used neural network
+        # If all intent confidence scores are beyond this threshold, set the current
+        # intent to `FALLBACK_INTENT_NAME`
+        THRESHOLD_KEY: DEFAULT_NLU_FALLBACK_THRESHOLD,
+        # The intent which is used to signal that the NLU confidence was below the
+        # threshold.
+        FALLBACK_INTENT_NAME_KEY: DEFAULT_NLU_FALLBACK_INTENT_NAME,
+    }
+
+    @classmethod
+    def required_components(cls) -> List[Type[Component]]:
+        return [IntentClassifier]
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        """Process an incoming message.
+
+        This is the components chance to process an incoming
+        message. The component can rely on
+        any context attribute to be present, that gets created
+        by a call to :meth:`rasa.nlu.components.Component.create`
+        of ANY component and
+        on any context attributes created by a call to
+        :meth:`rasa.nlu.components.Component.process`
+        of components previous to this one.
+
+        Args:
+            message: The :class:`rasa.nlu.training_data.message.Message` to process.
+
+        """
+
+        if not self._should_fallback(message):
+            return
+
+        message.data[INTENT_KEY] = self._fallback_intent()
+        message.data[INTENT_RANKING_KEY].insert(0, self._fallback_intent())
+
+    def _should_fallback(self, message: Message) -> bool:
+        return all(
+            intent_result.get(INTENT_CONFIDENCE_KEY, 0)
+            < self.component_config[THRESHOLD_KEY]
+            for intent_result in message.data[INTENT_RANKING_KEY]
+        )
```

Comment:
why don't we simply check that confidence of `INTENT_KEY`, which is the maximum intent is lower than threshold?